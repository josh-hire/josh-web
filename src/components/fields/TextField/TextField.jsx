import { InputBase as Base } from '@mantine/core';
import PropTypes from 'prop-types';

export default function TextField({
  disabled,
  icon,
  iconWidth,
  invalid,
  radius,
  rightSection,
  rightSectionWidth,
  ...rest
}) {
  return (
    <Base
      disabled={disabled}
      icon={icon}
      iconWidth={iconWidth}
      invalid={invalid}
      radius={radius}
      rightSection={rightSection}
      rightSectionWidth={rightSectionWidth}
      {...rest}
    />
  );
}

TextField.defaultProps = {
  disabled: false,
  icon: null,
  iconWidth: 36,
  invalid: false,
  radius: 3,
  rightSection: null,
  rightSectionWidth: 36,
};

TextField.propTypes = {
  disabled: PropTypes.bool,
  icon: PropTypes.node,
  iconWidth: PropTypes.number,
  invalid: PropTypes.bool,
  radius: PropTypes.number,
  rightSection: PropTypes.node,
  rightSectionWidth: PropTypes.number,
};
