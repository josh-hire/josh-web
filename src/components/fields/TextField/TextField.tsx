import { InputBase as Base, InputBaseProps } from '@mantine/core';
import React from 'react';

interface TextFieldProps extends InputBaseProps {
  icon?: React.ReactNode;
  iconWidth?: number;
  rightSection?: React.ReactNode;
  rightSectionWidth?: number;
}

const TextField: React.FC<TextFieldProps> = ({
  disabled = false,
  icon = null,
  iconWidth = 36,
  radius = 3,
  rightSection = null,
  rightSectionWidth = 36,
  ...rest
}) => {
  return (
    <Base
      disabled={disabled}
      icon={icon}
      iconWidth={iconWidth}
      radius={radius}
      rightSection={rightSection}
      rightSectionWidth={rightSectionWidth}
      {...rest}
    />
  );
};

export default TextField;
