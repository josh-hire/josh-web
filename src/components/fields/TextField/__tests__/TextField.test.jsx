import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import TextField from '../index';

describe('src/components/fields/TextField', ()=>{
  test('TextField',()=>{
    const { asFragment } = render(<TextField />);
    expect(asFragment()).toMatchSnapshot();
  });
});
