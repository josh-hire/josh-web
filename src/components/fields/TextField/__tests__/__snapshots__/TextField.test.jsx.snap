// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/TextField TextField 1`] = `
<DocumentFragment>
  <div
    class="mantine-InputWrapper-root mantine-InputBase-root mantine-1m3pqry"
  >
    <div
      class="mantine-Input-wrapper mantine-InputBase-wrapper mantine-12sbrde"
    >
      <input
        aria-invalid="false"
        class="mantine-Input-input mantine-InputBase-input mantine-1xsag3x"
        id="mantine-r0"
      />
    </div>
  </div>
</DocumentFragment>
`;
